src/state/reducers/buddies.ts

```diff
@@ -59,4 +61,30 @@ export const reducer: automaton.Reducer<State, actions.Action> = (
   }
 };
 
-export const getBuddies = ({ buddies }: types.AppState) => buddies;
+const getBuddiesWithStatus = (status: buddyApi.Buddy['status']) =>
+  flow(
+    ({ buddies }: types.AppState) => buddies,
+    RD.map(buddies =>
+      Object.values(buddies).filter(
+        ({ status: buddyStatus }) => buddyStatus === status,
+      ),
+    ),
+  );
```

Comment:
Good! Selectors best : )

Reply:
Imo, you should use getbuddies selector here to get the order right? Not sure which one is the right one but anyways, reuse the logic that already exists,